src/components/generic/InputField/InputField.jsx

```diff
@@ -112,8 +114,8 @@ export default function InputField({
             <Icon name={isPwdRevealed ? 'hide-password' : 'show-password'} />
           </Button>
         )}
-        {type !== 'password' && value !== undefined && (
-          <Button type="ghost" className={sc('input-field-right-icon')} size={28} onClick={clearInput}>
+        {withClear && value && (
```

Comment:
value !== undefined.

Also, the right padding of the input would need to be adjusted to allow space for the icon(s).

I would recommend creating a container for these absolutely positioned icons (clear and eye), putting a ref on it, calling useElementDimensions on it and updating the right padding of the input to the width of this icon container (plus some offset).